workers/unity/Assets/Gdk/Physics/Systems/ApplyTransformUpdatesSystem.cs

```diff
@@ -21,8 +23,28 @@ protected override void OnUpdate()
         {
             for (var i = 0; i < transformUpdateData.Length; i++)
             {
-                var newUpdates = transformUpdateData.TransformUpdate[i].Buffer;
-                transformUpdateData.BufferedTransform[i].TransformUpdates.AddRange(newUpdates);
+                var transformUpdates = transformUpdateData.TransformUpdate[i].Buffer;
+                var lastTransformSnapshot = transformUpdateData.BufferedTransform[i].LastTransformSnapshot;
+                var transformSnapshots = new List<SpatialOSTransform>();
+                foreach (var update in transformUpdates)
+                {
+                    if (update.Location.HasValue)
+                    {
+                        lastTransformSnapshot.Location = update.Location.Value;
+                    }
+
+                    if (update.Rotation.HasValue)
+                    {
+                        lastTransformSnapshot.Rotation = update.Rotation.Value;
+                    }
+
+                    if (update.Tick.HasValue)
+                    {
+                        lastTransformSnapshot.Tick = update.Tick.Value;
+                    }
```

Comment:
In the latest commit it seems you had removed the call to `transformSnapshots.Add()`

Do we need to add to `transformSnapshots` for every `update`, or only `update` which has one or more of these three values modified?

Reply:
Holy shit, I derped here. Yes, that line wasn't supposed to be removed.